srcpkgs/mimalloc/template

```diff
@@ -0,0 +1,31 @@
+# Template file for 'mimalloc'
+pkgname=mimalloc
+version=1.7.2
+revision=1
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=RelWithDebInfo"
```

Comment:
No, please!

Reply:
Please patch their build system to remove the addition of `-none` non-sense.

Reply:
Is it neccessary to upstream the patch? Will it be OK to temporary have patch in `patches` until they accept it?

Reply:
@sgn why does `common/build-style/cmake.sh` set `CMAKE_BUILD_TYPE` to `None` though? Isn't `RelWithDebInfo` more suitable (so it's release build, but with debug info which can be used to create `-dbg` package)?

Reply:
- We don't want the flags KitWare implies (`-DNDEBUG`)
- racer wants their `-O3`
- We allow build without debug, too. And sometime, `-g` is unmanagable, look at `llvm12` ;)
- broken build systems are broken build systems, to them, there are only `Debug` and `Release`

Reply:
@sgn what is KitWare?

Reply:
Google is your friend ;)

Reply:
@sgn adding "cmake" to my search query worked.
TLDR; it's the company behind CMake